aws_sagemaker_studio/sagemaker_studio_image_build/tensorflow_bring_your_own/tensorflow_bring_your_own.ipynb

```diff
@@ -0,0 +1,740 @@
+    "\n",
+    "Docker uses a simple file called a `Dockerfile` to specify how the image is assembled. An example is provided below. You can build your Docker images based on Docker images built by yourself or by others, which can simplify things quite a bit.\n",
+    "\n",
+    "Docker has become very popular in programming and devops communities due to its flexibility and its well-defined specification of how code can be run in its containers. It is the underpinning of many services built in the past few years, such as [Amazon ECS].\n",
+    "\n",
+    "Amazon SageMaker uses Docker to allow users to train and deploy arbitrary algorithms.\n",
+    "\n",
+    "In Amazon SageMaker, Docker containers are invoked in a one way for training and another, slightly different, way for hosting. The following sections outline how to build containers for the SageMaker environment.\n",
+    "\n",
+    "Some helpful links:\n",
+    "\n",
+    "* [Docker home page](http://www.docker.com)\n",
+    "* [Getting started with Docker](https://docs.docker.com/get-started/)\n",
+    "* [Dockerfile reference](https://docs.docker.com/engine/reference/builder/)\n",
+    "* [`docker run` reference](https://docs.docker.com/engine/reference/run/)\n",
+    "\n",
+    "[Amazon ECS]: https://aws.amazon.com/ecs/\n",
+    "\n",
+    "### How Amazon SageMaker runs your Docker container\n",
+    "\n",
+    "Because you can run the same image in training or hosting, Amazon SageMaker runs your container with the argument `train` or `serve`. How your container processes this argument depends on the container.\n",
+    "\n",
+    "* In this example, we don't define an `ENTRYPOINT` in the Dockerfile so Docker runs the command [`train` at training time](https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-training-algo.html) and [`serve` at serving time](https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-inference-code.html). In this example, we define these as executable Python scripts, but they could be any program that we want to start in that environment.\n",
+    "* If you specify a program as an `ENTRYPOINT` in the Dockerfile, that program will be run at startup and its first argument will be `train` or `serve`. The program can then look at that argument and decide what to do.\n",
+    "* If you are building separate containers for training and hosting (or building only for one or the other), you can define a program as an `ENTRYPOINT` in the Dockerfile and ignore (or verify) the first argument passed in. \n",
+    "\n",
+    "#### Running your container during training\n",
+    "\n",
+    "When Amazon SageMaker runs training, your `train` script is run, as in a regular Python program. A number of files are laid out for your use, under the `/opt/ml` directory:\n",
+    "\n",
+    "    /opt/ml\n",
+    "    |-- input\n",
```

Comment:
Should mention the /code directory and how the dockerfile will copy the training and hosting scripts in there.